src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -70,98 +80,111 @@ def setUp(self):
 
         self.tempdir = self.client_fixture.tempdir
         self.node_directory = self.client_fixture.node_directory
-        self.config_dir = FilePath(self.mktemp())
-        create_global_configuration(self.config_dir, u"tcp:4321", self.node_directory, u"tcp:localhost:4321")
+
+        # the Web APIs need a reference to a "global_service" .. which
+        # is cli.MagicFolderService (a MultiService in fact). It
+        # doesn't declare an interface, but only uses
+        # "get_folder_service(folder_name)" .. so we'll duck-type it
+        # instead
+
+        # inside create_testing_configuration, the GlobalService is
+        # hooked up to an in-process HTTP API root which is
+        # interrogated for information. So, we want to control which
+        # magic-folders it sees
+
+        self.magic_folders = {}
+
+        class GlobalService(object):
+            def get_folder_service(s, name):
+                return self.magic_folders[name]
+
+        self.config = create_testing_configuration(
+            FilePath(self.mktemp()),
+            self.node_directory,
+            GlobalService(),
+        )
 
     @defer.inlineCallbacks
     def test_list_none(self):
         """
         When there are no Magic Folders at all, the output of the list command
         reports this.
         """
-        outcome = yield cli(
-            self.config_dir,
-            [b"list"],
+        output = StringIO()
```

Comment:
This is supposed to be like sys.stdout right?  `BytesIO` might be more realistic.  Although if this is just mirroring what `cli` was doing then nevermind, let's not blow up the scope of this PR.